component/module/file/file_test.go

```diff
@@ -56,6 +60,12 @@ func TestModule(t *testing.T) {
 				Registerer:    prometheus.NewRegistry(),
 				OnStateChange: func(e component.Exports) {},
 				DataPath:      t.TempDir(),
+				ModuleSystem: flow.NewModuleSystem(&flow.ModuleOptions{
+					Logger:    l,
+					Tracer:    trace.NewNoopTracerProvider(),
+					Clusterer: cl,
+					Reg:       prometheus.DefaultRegisterer,
+				}),
 			}
```

Comment:
Like with pkg/module before, I don't want `flow.NewModuleSystem` to be a part of our public API at all:

1. Components should be using `pkg/flow/componenttest` to test their implementations, where `pkg/flow/componenttest` can return some minimally useful interface specific for testing. 
2. Tests against the actual module code should be moved to pkg/flow rather than be done in individual module loaders.

Reply:
Rewrote the tests to pass through flow instead of directly instantiating the component, for file_test it needs the real module loader. Checks most of the same things but now has to go through the ComponentInfos call to get information on the component.

Reply:
> file_test it needs the real module loader

Can you explain why it needs the real module loader?

Reply:
It has a few tests that extract handle bad module loading. Its possible those arent the right place for the tests but I didnt want to make that determination in this PR.